lib/state/base_view_model.dart

```diff
@@ -0,0 +1,29 @@
+import 'package:flutter/widgets.dart';
+
+class BaseViewModel<T> extends ChangeNotifier {
```

Comment:
Why not use the base from Stacked itself?

Reply:
It wasn't designed to have immutable state. I'll look into it more.